react/week3/src/components/TodoItem.js

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import EditTodoForm from "./EditTodoForm";
+import PropTypes from 'prop-types';
+
+function TodoItem({

```

Comment:
The editState works but its not initially clear how. Remember you have to make your code and your intentions clear to others. 

I suggest rewriting the TodoItem like this:
```
function TodoItem({
  id,
  description,
  deadline,
  completed,
  removeTodo,
  editTodo,
  toggleTodo
}) {
  const [state, editState] = useState(false);

  const toggleEditState = () => editState(prev => !prev);
  
  const editFormShow = () => {
    return (<EditTodoForm
          editTodo={editTodo}
          id={id}
          description={description}
          toggleEditState={toggleEditState}
      />)
  }

  const editFormHide = () => {
    return (<div>
      {description} {deadline}
      <input
        type="checkbox"
        name="completed"
        value={completed}
        onChange={() => toggleTodo(id)}
      />
      <button onClick={() => removeTodo(id)}>Delete</button>
      <button onClick={toggleEditState}>Edit</button>
    </div>)
  }

  return (
    <li style={{ textDecoration: completed ? "Line-through" : "none" }}>
      {state ? editFormShow() : editFormHide() }
    </li>
  );
}
```

Reply:
thanks for comments, I will make the changes and merge the branch